element_facemap/facial_behavior_estimation.py

```diff
@@ -259,111 +275,110 @@ class FacialSignal(dj.Imported):
+        movmask             : longblob    # PC (y, x)
+        projection          : longblob    # projections onto the principle component (nframes)
         """
 
     class Summary(dj.Part):
         definition = """
         -> master
         ---
-        sbin                : int           # spatial bin size
-        avgframe            : longblob      # 2d np.array - average binned frame
-        avgmotion           : longblob      # 2d nd.array - average binned motion frame
+        sbin          : int         # spatial bin size
+        avgframe      : longblob    # 2d np.array - average binned frame
+        avgmotion     : longblob    # 2d nd.array - average binned motion frame
         """
 
-
-    def get_ncomponents(self, Svs, threshold=0.95):
-        # Calculate the number of PCA components that will make up to the first 95% variance.
-        squared_Svs = Svs ** 2
-        cumulative_explained_variances = np.cumsum(squared_Svs / sum(squared_Svs))
-        return sum(cumulative_explained_variances < threshold)
-
-
     def make(self, key):
-        dataset, creation_time = get_loader_result(key, FacemapTask)
-        params = (FacemapTask & key).fetch1('facemap_params')
-        print('key:', key)
-        print('params', params)
+        dataset, _ = get_loader_result(key, FacemapTask)
 
-        self.Region.insert([
-            dict(
-                key,
-                roi_no=i,
-                xrange=dataset['rois'][i]['xrange'],
-                yrange=dataset['rois'][i]['yrange'],
-                xrange_bin=dataset['rois'][i]['xrange_bin'],
-                yrange_bin=dataset['rois'][i]['yrange_bin']
-            ) for i in range(1, len(dataset['rois']))
-        ])
+        self.insert1(key)
 
-        # MotionSVD
-        #do_mot_svd = params.keys() # (FacemapTask & key).fetch1('do_mot_svd')
-        if 'motSv' in params.keys():
-            n_components = self.get_ncomponents(dataset['motSv'])
-            for roi_no in range(len(dataset['rois'])):
-                self.MotionSVD.insert(
-                    dict(
-                        key,
-                        roi_no=roi_no,
-                        singular_value=dataset['motSv'][i],
-                        motmask=dataset['motMask_reshape'][roi_no+1][i],
-                        projection=dataset['motMask_reshape'][roi_no+1][:,:,i],
-                    ) for i in range(n_components)
+        self.Region.insert(
+            [
+                dict(
+                    key,
+                    roi_no=i,
+                    xrange=dataset["rois"][i]["xrange"],
+                    yrange=dataset["rois"][i]["yrange"],
+                    xrange_bin=dataset["rois"][i]["xrange_bin"],
+                    yrange_bin=dataset["rois"][i]["yrange_bin"],
+                    motion=dataset["motion"][i],
                 )
+                for i in range(len(dataset["rois"]))
+            ]
+        )
```

Comment:
```suggestion
        self.Region.insert(
            [
                dict(
                    key,
                    roi_no=i,
                    motion=motion,
                    **roi
                )
                for i, (roi, motion) in enumerate(zip(dataset["rois"], dataset['motion']))
            ]
        )
```

Reply:
`dataset["roi"]` contains a lot of keys. `xrange`, `yrange`, `xrange_bin`, and `yrange_bin` have to fetched one by one. This suggestion is not acceptable.